### ISSUE TITLE
Project still alive ?

### ISSUE THREAD

**Issue by tobixx** [calston/riakasaurus#34]
Hi Calston,

due too the (non) activity, the build status and open issues  I want to check If you are still in charge of this project and if you would be open for continuation of it.

I wanted to implement a high performance multi head pbc io loop, but then I saw the "rusty" state of the project. 

Because there are such big changes needed, to get back on track with the Riak development, before I could start to work on my implementation (deprecate/drop http support - just as one example), I want to check out the perspective  from your view first.

Please give me a life sign ;-)

**Comment by calston** [comment]
Well I'm still alive but the commercial drivers behind this are no longer. I'm definitely keen to get it back in shape but the API instability moving to 2.0 got annoying. I have been meaning to check if that has gotten better

**Comment by tobixx** [comment]
Great to hear! Thanks for the information.

Maybe you are interested in the comments [here](https://github.com/eminarcissus/riakasaurus/commit/0cad396f036d304fb7b296203aef2a8e91f8f3a2#commitcomment-20144283) too.

Now that I know that you are still interested in it - I can consider it more reliable for future use.

And yes - I agree - the changes they have made are ... tough ... if you follow more the "never change a running system approach" and wake up some day where most of the things gotten deprecated already ^^ .

While looking into the head implementation, again, I thought about a implementation for http also (in the [official](https://github.com/basho/riak-python-client/pull/502) client). My surprise was huge while I searched the documentation for it: first I could not find any reference to the HEAD request anymore, second I noticed that the whole http protocol is deprecated.

Luckily for us that we provided the pbc protocol years ago. Saying that, I want to mention that we do not depend on the http implementation. So my first thought was to just remove it to get at least started to make Jenkins happy again. 

Anyway, now that you are back on track, I would suggest that you maybe write about your plans and directions once you re-checked the current api changes. 

Maybe we could help out here and there ...

**Comment by calston** [comment]
Good idea, I think it probably needs a total rewrite though, the state machine in PBC is needlessly complex. It would be ideal if the official python library was more loosely coupled to allow dropping Twisted into it.

**Comment by tobixx** [comment]
**Attention** : some crazy (without any proof) idea follows:

I while ago I [read](https://glyph.twistedmatrix.com/2014/05/the-report-of-our-death.html) glyphs opinion about asyncio and there was again a [talk](https://www.youtube.com/watch?v=82vuCZ4FLFE) on Pycon this year about this topic and in the last [changelog](http://labs.twistedmatrix.com/2016/11/twisted-166-released.html) ther is following mentioned:

> Fixes for async/await & twisted.internet.defer.ensureDeferred, meaning it's getting closer to prime time!

Knowing that asyncio alternatives [exists](https://github.com/rambler-digital-solutions/aioriak) and seeing that other [projects](https://pypi.python.org/pypi/autobahn) already support mixed Twisted/asyncio operation and finding things like [txaio](https://github.com/crossbario/txaio), maybe is worth a thought on hooking onto such alternative instead of re-inventing a pure twisted implementation.
